src/main/java/com/atlassian/jira/cloud/jenkins/listeners/ConfigurationSaveableListener.java

```diff
@@ -1,41 +1,123 @@
 import java.util.logging.Logger;
 
+import static com.atlassian.jira.cloud.jenkins.util.IpAddressProvider.getIpAddress;
+
 @Extension
 public class ConfigurationSaveableListener extends SaveableListener {
+    private static final String SITE_CONFIGURATION_TAG = "atl-jsw-site-configuration";
+    private final Logger logger = Logger.getLogger(ConfigurationSaveableListener.class.getName());
 
-    private static final Logger LOGGER =
-            Logger.getLogger(ConfigurationSaveableListener.class.getName());
+    private PluginConfigApi pluginConfigApi;
+    private SecretRetriever secretRetriever;
+    private XmlUtils xmlUtils;
 
-    private transient PluginConfigApi pluginConfigApi;
+    @Inject
+    public void setSecretRetriever(final SecretRetriever secretRetriever) {
+        this.secretRetriever = secretRetriever;
+    }
 
     @Inject
     public void setConnectionDataApi(final PluginConfigApi pluginConfigApi) {
         this.pluginConfigApi = pluginConfigApi;
     }
 
+    @Inject
+    public void setXmlUtils(final XmlUtils xmlUtils) {
+        this.xmlUtils = xmlUtils;
+    }
+
     @Override
     public void onChange(final Saveable saveable, final XmlFile file) {
-        if (!(saveable instanceof com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig)) {
-            return;
+        if (saveable instanceof com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig) {
+            sendPluginConfigData(file);
+        }
+    }
+
+    void sendPluginConfigData(final XmlFile file) {
+        try {
+            Element rootElement = this.xmlUtils.parseXmlFile(file);
+            List<PluginSiteData> siteDataList = createSiteDataList(rootElement);
+            String autoBuildsRegex = this.xmlUtils.extractXmlValue(rootElement, "autoBuildsRegex");
+            boolean autoBuildsEnabled =
+                    this.xmlUtils.extractBooleanValue(rootElement, "autoBuildsEnabled");
+            boolean autoDeploymentsEnabled =
+                    this.xmlUtils.extractBooleanValue(rootElement, "autoDeploymentsEnabled");
+            String autoDeploymentsRegex =
+                    this.xmlUtils.extractXmlValue(rootElement, "autoDeploymentsRegex");
+
+            for (PluginSiteData data : siteDataList) {
+                Optional<String> maybeSecret =
+                        secretRetriever.getSecretFor(data.getCredentialsId());
+                maybeSecret.ifPresent(
+                        secret ->
+                                sendDataToPluginApi(
+                                        data,
+                                        autoBuildsRegex,
+                                        autoBuildsEnabled,
+                                        autoDeploymentsEnabled,
+                                        autoDeploymentsRegex,
+                                        secret));
+            }
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Error processing plugin configuration data", e);
```

Comment:
I don't think we should throw a runtime exception here as we'll lose info about the original exception and can lead to unexpected behaviour (including potential crashes 😬 ) I just did some reading on common XML errors Maybe we can do this a little more gracefully by doing something like:

```
catch (ParserConfigurationException e) {
        // Handle XML parsing configuration error
        logger.log(Level.SEVERE, "XML parsing configuration error", e);
        // we could just return here or return a default value?
        return;
        } catch (SAXException e) {
        // Handle XML parsing SAX (simple api for xml) error
        logger.log(Level.SEVERE, "XML parsing SAX error", e);
        // same as above
        return;
        } catch (Exception e) {
        // Handle other unexpected exceptions
        logger.log(Level.SEVERE, "Error processing plugin configuration data", e);
        // Custom error handling class (you can call it whatever you want) that would extend from Exception
        throw new XmlProcessingException("Failed to process plugin configuration data", e);
        }
    }
```

Reply:
Agree with not throwing runtime error for exactly what you said!!!
But since we aren't doing anything meaningful with various exceptions I've kept it generic, all we want to do is log, I've avoided re throwing the error as well for the same reason, if an error occurs we don't want to propagate it up, logging is as much dealing with it as we need and higher functions will just muddy the stacktrace.